exec-workitem/src/main/java/org/jbpm/process/workitem/exec/ExecShellScriptWorkItemHandler.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.process.workitem.exec;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.jbpm.process.workitem.core.AbstractLogOrThrowWorkItemHandler;
+import org.jbpm.process.workitem.core.util.RequiredParameterValidator;
+import org.jbpm.process.workitem.core.util.Wid;
+import org.jbpm.process.workitem.core.util.WidMavenDepends;
+import org.jbpm.process.workitem.core.util.WidParameter;
+import org.jbpm.process.workitem.core.util.WidResult;
+import org.jbpm.process.workitem.core.util.service.WidAction;
+import org.jbpm.process.workitem.core.util.service.WidAuth;
+import org.jbpm.process.workitem.core.util.service.WidService;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * WorkItemHandler that is capable of executing shell script file
+ * <ul>
+ * <li>ShellScriptLocation - Absolute or relative path of shell script file - mandatory</li>
+ * </ul>
+ */
+
+
+@Wid(widfile = "ExecShellScriptDefinitions.wid", name = "ExecShellScript",
+        displayName = "ExecShellScript",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.exec.ExecShellScriptWorkItemHandler()",
```

Comment:
First question that comes to mind is - how is this different than the existing exec workitem  - https://github.com/kiegroup/jbpm-work-items/tree/master/exec-workitem

If they are the same thing, can the existing one be enhanced to do file support like what it seems you need here? Do we need two handlers??

Reply:
I had thought of enhancing the existing execute work item but felt that using processBuilder approach was better than commandLine executor for executing shell scripts. Enhancing existing workitem using processBuilder for executing file seemed not appropriate. I observed few other WIH handlers like GIT, twitter etc where separate handlers was created for separate operations. So felt that exceuting a command vs executing file would be separate operations, so decided with separate handler. It's the first time I am contributing to the open source repo, so I am not very sure about design details. Let me know if you have any recommendations.

Reply:
I think you make fair points on possibly having two separate handlers. thanks